src/Interpreters/ActionsDAG.cpp

```diff
@@ -596,6 +596,34 @@ void ActionsDAG::removeUnusedActions(const std::unordered_set<const Node *> & us
     std::erase_if(inputs, [&](const Node * node) { return !visited_nodes.contains(node); });
 }
 
+
+void ActionsDAG::removeAliasesForFilter(const std::string & filter_name)
+{
+    const auto & filter_node = findInOutputs(filter_name);
+    std::stack<Node *> stack;
+    stack.push(const_cast<Node *>(&filter_node));
+
+    std::unordered_set<const Node *> visited;
+    visited.insert(stack.top());
+
+    while (!stack.empty())
+    {
+        auto * node = stack.top();
+        stack.pop();
+        for (auto & child : node->children)
+        {
+            while (child->type == ActionType::ALIAS)
```

Comment:
I remember that we discusses that 'unrolling' aliases can be beneficial for any expression step. 
But this method cannot be applied as it is because we have to consider aliases that are needed to change names in output, right?
Can't filter name be changed after applying this method?

Reply:
I suppose filter column name does not change.
But I agree, that it'd be nice if we remove all unnecessary alias nodes from dag.

Reply:
In this case, the names of outputs are not changed at all, cause here we just replace
`f(x) -> y alias x -> y` chain to direct `f(x) -> y`